src/html/fetch-markdown.js

```diff
@@ -10,102 +10,57 @@
  * governing permissions and limitations under the License.
  */
 const client = require('request-promise-native');
-const URI = require('uri-js');
+const { iter } = require('@adobe/helix-shared').sequence;
 const { bail } = require('../helper');
-
-function uri(root, owner, repo, ref, path) {
-  const rootURI = URI.parse(root);
-  const rootPath = rootURI.path;
-  // remove double slashes
-  const fullPath = `${rootPath}/${owner}/${repo}/${ref}/${path}`.replace(
-    /\/+/g,
-    '/',
-  );
-
-  return URI.serialize({
-    scheme: rootURI.scheme,
-    host: rootURI.host,
-    port: rootURI.port,
-    path: fullPath,
-  });
-}
+const { normalizeURL } = require('../utils/util');
 
 /**
- * Fetches the Markdown specified in the action and returns
- * the body of the Markdown document
- * @param {import("../context").Context} ctx some param
- * @param {import("../context").Action} action some other param
+ * Stage of the markdown->html pipeline that actually fetches the content.
+ *
+ * @param {import("../context").Context} ctx
+ * @param {import("../context").Action} action
  */
-async function fetch(
-  { content: { sources = [] } = {} },
-  {
-    secrets = {},
-    request,
-    logger,
-  },
-) {
+const fetch = async ({ content = [] }, { request, logger }) => {
+  const { sources = [] } = content;
+
+  logger.info(`A`);
   if (!request || !request.params) {
     return bail(logger, 'Request parameters are missing');
   }
 
-  let timeout;
-  if (!secrets.HTTP_TIMEOUT) {
-    logger.warn('No HTTP timeout set, risk of denial-of-service');
-  } else {
-    timeout = secrets.HTTP_TIMEOUT;
-  }
-
-  // get required request parameters
-  const {
-    owner, repo, path,
-  } = request.params;
+  const { owner, repo, path } = request.params;
+  const ref = request.params.ref || 'refs/heads/master';
+  const dump = (o) => require('util').inspect(o, {depth: null});
 
-  let { ref } = request.params;
+  logger.info(`B ${dump(request)}`);
+  logger.info(`X ${dump({owner, repo, path, ref})}`);
+  // for (const [key, val] of iter({ owner, repo, path })) {
+  //   if (!val) {
+  //     return bail(logger, `Unknown ${key}, cannot fetch content.`);
+  //   }
+  // }
 
-  // bail if a required parameter cannot be found
-  if (!owner) {
-    return bail(logger, 'Unknown owner, cannot fetch content');
-  }
-  if (!repo) {
-    return bail(logger, 'Unknown repo, cannot fetch content');
-  }
-  if (!path) {
-    return bail(logger, 'Unknown path, cannot fetch content');
-  }
-  if (!ref) {
-    logger.warn(`Recoverable error: no ref given for ${repo}/${owner}.git${path}, falling back to master`);
-    ref = 'master';
-  }
+  logger.info(`C`);
+  const opt = {
+    url: normalizeURL(`https://api.github.com/repos/${owner}/${repo}/contents/${path}`),
+    qs: { ref },
+    headers: {
+      Accept: 'application/vnd.github.VERSION.raw"',
```

Comment:
```suggestion
      Accept: 'application/vnd.github.VERSION.raw',
```

I think you have a superfluous `"` in there.